### PR TITLE
10 menu feature 2 game creation and file reading

### PR DESCRIPTION
**Pull request by jchenDDD**
applied changes requested from the last pull. Changes:
- FileReader
- GameData
- data_base
- Added Player factory

### CODE REVIEW

**Comment by stalight** [comment]
You might want to pull from main for Collectibles, Player and Weapon, Player Factory can be changed if you did anything to it

File: data_base/Player.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        "weaponAttackPoint": 1223
+      },
+      "armor": {
+        "name": "armor",
+        "dmgReduced": 122
+      }
+    },
+    "location": [0,0]
+  }
+]
```
Comment (HYBleek) [inline data_base/Player.json:28]: Is this an example of the player? Or the player start with the stats?

**Review by HYBleek — COMMENTED** [review]
I do not really know how the JSON work, but the builder part is good

File: data_base/Player.json (hunk 1 above)
Comment (jchenDDD) [reply data_base/Player.json:28]: this is just an example, you guys can tell me the starting stats or change it yourself.

**Review by villanueva-j — DISMISSED** [review]
Is it okay to directly call the object constructors in you deserialization classes or should you use the respective factory?

**Comment by villanueva-j** [comment]
you need me and jillian to finish the databases for monster and equipment in order to implement the deserialization casses for them right?

**Comment by jchenDDD** [comment]
> 
directing calling new in a factory or calling it in deserialization does not really have a difference I think. In the end, the factory's job is to create instances of objects. I guess you can view deserializations as helper classes of factories.

**Comment by jchenDDD** [comment]
> you need me and jillian to finish the databases for monster and equipment in order to implement the deserialization casses for them right?

Not really. All I need is to see the Monster constructor and deserialize by assuming that the keys in your json are spelled **EXACTLY** the same as their corresponding parameter name in the constructor.

**Comment by villanueva-j** [comment]
> > you need me and jillian to finish the databases for monster and equipment in order to implement the deserialization casses for them right?
> 
> Not really. All I need is to see the Monster constructor and deserialize by assuming that the keys in your json are spelled **EXACTLY** the same as their corresponding parameter name in the constructor. See my player.json to see how a key refers to another object.

oh okay, are you gonna implement those ones later then?

File: src/main/java/usecase_gamedata/PlayerFactory.java [hunk 2]
```diff
@@ -1,25 +1,32 @@
...
+        // flexible on that. For example, have a new input of String class and for every class the factory
+        // gets the starter equipment and items for that specific class and build the player.
 
 
-public class PlayerFactory {
-    private PlayerBuilder builder;
-    private JSONObject object = Json.createObject
 
-    public Player create(JSONObject jsonObject, GameData gameData) {
-        Player player;
```
Comment (HongyiZhan) [inline src/main/java/usecase_gamedata/PlayerFactory.java:15]: I am very confused about this deletion. Could you explain this?

**Review by HongyiZhan — CHANGES_REQUESTED** [review]

File: src/main/java/usecase_gamedata/PlayerFactory.java (hunk 2 above)
Comment (jchenDDD) [reply src/main/java/usecase_gamedata/PlayerFactory.java:15]: GameFileReader no longer returns an jsonObject, it returns the string form of jsonObject. We use the string form of the jsonObject to make Player.

**Comment by HongyiZhan** [comment]
How to use the Jason file from database

**Review by HongyiZhan — DISMISSED** [review]
ok, now the Jason file is work

**Comment by stalight** [comment]
You probably don't want to merge gradle, it'll cause conflicts

**Comment by stalight** [comment]
You can delete it in the file changed

**Review by HongyiZhan — APPROVED** [review]
text info can be read into the interface.

**Comment by HongyiZhan** [comment]
how do you control the player hp value

**Review by UTDQi — APPROVED** [review]

**Review by villanueva-j — APPROVED** [review]

**Review by HYBleek — APPROVED** [review]

**Pull request merged**